utils/pub-helpers.rkt

```diff
@@ -0,0 +1,135 @@
+#lang racket
+
+(provide
+ ;; (Union String ByteString) -> (Listof publication)
+ ;;
+ ;; Given the path portion of a URI for an author's XML file on DBLP some author as a string or
+ ;; byte-string (e.g. "/pers/xx/s/Shivers:Olin.xml"), returns a list of publication structs for all
+ ;; the publications on that DBLP page we are able to read (currently supports article, book, and
+ ;; inproceedings types).
+ scrape-dblp
+
+ ;; (Listof publication) output-port -> void
+ ;;
+ ;; Writes the given publications to the given port in a format that can easily be copied and pasted
+ ;; into publications.rkt
+ write-pubs)
+
+;; ---------------------------------------------------------------------------------------------------
+
+(require xml
+         net/http-client
+         "../publication-data.rkt")
+
+(define (write-pubs pubs port)
+  (for ([pub pubs])
+  (match-define (publication title authors venue year link) pub)
+  (fprintf port
+"(publication ~s
+             ~s
+             ~s
+             ~s
+             ~s)\n"
+           title authors venue year link)))
+
+;; ---------------------------------------------------------------------------------------------------
+;; DBLP
+
+(define supported-record-types `(article book inproceedings))
+
+(define (scrape-dblp uri)
+  (define-values (status headers xml-stream) (http-sendrecv "dblp.uni-trier.de" uri))
+  (define pub-records
+    (map (compose1 first element-content)
+         (get-children-of-name (document-element (read-xml xml-stream)) 'r)))
+  (for/list ([record pub-records] #:when (member (record-type record) supported-record-types))
+    (dblp-record->pub record)))
+
+(define (dblp-record->pub record)
+  (define authors
+    (map (compose1 pcdata-string first element-content) (get-children-of-name record 'author)))
+  (define author-text
```

Comment:
Guess it's late now, but Scribble has an [`authors`](http://docs.racket-lang.org/scriblib/autobib.html?q=authors#%28def._%28%28lib._scriblib%2Fautobib..rkt%29._authors%29%29) function.